推送配置过程中，删除ingress，higress-controller会panic重启

**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via [ASRC](https://security.alibaba.com/)(Alibaba Security Response Center) where the issue will be triaged appropriately.**

- [ ] I have searched the [issues](https://github.com/alibaba/higress/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
推送配置过程中，删除ingress，higress-controller会panic重启
higress-core日志

```
2024-08-14T12:40:35.602182Z     error   klog    Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 262 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x29a1360?, 0x5109a70})
        k8s.io/apimachinery@v0.24.1/pkg/util/runtime/runtime.go:74 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00322fba0?})
        k8s.io/apimachinery@v0.24.1/pkg/util/runtime/runtime.go:48 +0x75
panic({0x29a1360, 0x5109a70})
        runtime/panic.go:884 +0x212
github.com/alibaba/higress/pkg/ingress/kube/kingress.(*controller).onEvent(0xc0009b66e0, {{0xc0010aabd9?, 0xc00015cf00?}, {0xc0011c0940?, 0x20?}})
        github.com/alibaba/higress/pkg/ingress/kube/kingress/controller.go:166 +0xa9
github.com/alibaba/higress/pkg/ingress/kube/kingress.(*controller).processNextWorkItem(0xc0009b66e0)
        github.com/alibaba/higress/pkg/ingress/kube/kingress/controller.go:154 +0xed
github.com/alibaba/higress/pkg/ingress/kube/kingress.(*controller).worker(0xc0019c2ea0?)
        github.com/alibaba/higress/pkg/ingress/kube/kingress/controller.go:143 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x1?)
        k8s.io/apimachinery@v0.24.1/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x27ba520?, {0x34b2b40, 0xc0039deed0}, 0x1, 0xc0001159e0)
        k8s.io/apimachinery@v0.24.1/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x1?, 0x443d85?)
        k8s.io/apimachinery@v0.24.1/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0xc0001159e0?, 0x0?)
        k8s.io/apimachinery@v0.24.1/pkg/util/wait/wait.go:90 +0x25
created by github.com/alibaba/higress/pkg/ingress/kube/kingress.(*controller).Run
        github.com/alibaba/higress/pkg/ingress/kube/kingress/controller.go:138 +0x2bb
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x25d33e9]
```

代码panic位置
![2024-08-15_10-16](https://github.com/user-attachments/assets/1450fbcf-2e9f-4a9d-9498-5c9d81bd4c16)

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```


### Ⅲ. Describe what you expected to happen


### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅴ. Anything else we need to know?


### Ⅵ. Environment:

- Higress version:  1.3.6，其他版本代码也有类似问题
- OS : linux
- Others:

--- a/pkg/ingress/kube/kingress/controller.go
+++ b/pkg/ingress/kube/kingress/controller.go
@@ -163,7 +163,6 @@ func (c *controller) processNextWorkItem() bool {
 func (c *controller) onEvent(namespacedName types.NamespacedName) error {
 	event := model.EventUpdate
 	ing, err := c.ingressLister.Ingresses(namespacedName.Namespace).Get(namespacedName.Name)
-	ing.Status.InitializeConditions()
 	if err != nil {
 		if kerrors.IsNotFound(err) {
 			event = model.EventDelete
@@ -180,6 +179,8 @@ func (c *controller) onEvent(namespacedName types.NamespacedName) error {
 	if ing == nil {
 		return nil
 	}
+
+	ing.Status.InitializeConditions()
 
 	// we should check need process only when event is not delete,
 	// if it is delete event, and previously processed, we need to process too.

